File: referralhero/src/main/java/com/sdk/rh/DeviceInfo.java
package com.sdk.rh;


import android.content.Context;

/**
 * <p>
 * Class is  responsible for capturing device info and updating
 * device info to ReferralHero requests
 * </p>
 */
public class DeviceInfo {

    private final Context context_;


    /**
     * Get the singleton instance for this class
     *
     * @return {@link DeviceInfo} instance if already initialised or null
     */
    static DeviceInfo getInstance() {
        RH b = RH.getInstance();
        if (b == null) return null;
        return b.getDeviceInfo();
    }

    DeviceInfo(Context context_) {
        this.context_ = context_;
    }
}

File: referralhero/src/main/java/com/sdk/rh/DeviceInfo.kt
package com.sdk.rh

import android.content.Context

/**
 *
 *
 * Class is  responsible for capturing device info and updating
 * device info to ReferralHero requests
 *
 */
class DeviceInfo internal constructor(private val context_: Context) {
    companion object {
        /**
         * Get the singleton instance for this class
         *
         * @return [DeviceInfo] instance if already initialised or null
         */
        val instance: DeviceInfo?
            get() {
                val b: RH = RH.instance ?: return null
                return b.deviceInfo
            }
    }
}